File: app/src/main/java/org/deiverbum/app/data/entity/LiturgyGroupEntity.java
package org.deiverbum.app.data.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = "liturgy_group", indices = {@Index(value = {"liturgyFK"},unique =
        true)},
        foreignKeys =
                {
        @ForeignKey(
                entity = LiturgiaEntity.class,
                parentColumns = "liturgyID",
                childColumns = "liturgyFK",
                onDelete = ForeignKey.CASCADE,
                onUpdate = ForeignKey.CASCADE)})
public class LiturgyGroupEntity {
    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "groupID")
    public Integer groupID=0;

    @NonNull
    @ColumnInfo(name = "liturgyFK")
    public Integer liturgyFK=0;

}

File: app/src/main/java/org/deiverbum/app/data/entity/LiturgyGroupEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.*
import androidx.room.ForeignKey.Companion.CASCADE

/**
 * Entidad para la tabla **`liturgy_group`** de la base de datos, que se ocupa de gestionar los grupos de Liturgia.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(
    tableName = "liturgy_group",
    indices = [Index(value = ["liturgyFK"], unique = true)],
    foreignKeys = [ForeignKey(
        entity = LiturgyEntity::class,
        parentColumns = arrayOf("liturgyID"),
        childColumns = arrayOf("liturgyFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    )]
)
data class LiturgyGroupEntity(
    @PrimaryKey
    @ColumnInfo(name = "groupID")
    val groupID: Int,

    @ColumnInfo(name = "liturgyFK")
    val liturgyFK: Int
)